include/dynd/callable.hpp

```diff
@@ -155,9 +155,9 @@ namespace nd {
 
     std::intptr_t get_narg() const { return get_type()->get_npos(); }
 
-    const ndt::type &get_arg_type(std::intptr_t i) const { return get_type()->get_pos_type(i); }
+    const ndt::type get_arg_type(std::intptr_t i) const { return get_type()->get_pos_type(i); }
 
-    const array &get_arg_types() const { return get_type()->get_pos_types(); }
+    nd::array get_arg_types() const { return get_type()->get_pos_types(); }
```

Comment:
Nothing prevents the tuple from disappearing, that's true. It's the responsibility of the code getting that property to make sure the lifetime of that reference is not longer than the reference of the object it came from.

This is pretty common - e.g. in std::vector, operator[] returns a `T&` reference with exactly the same considerations.